File: PHP/DesignPatterns/Observer.php
<?php

// Stock ticker observer:

class TickerObserver
{
    // The symbol that this observer will monitor for:
    private $symbol;
    // The last price observed:
    private $lastPrice;

    function __construct($symbol)
    {
        $this->symbol = $symbol;
        $this->lastPrice = 0.0;
    }

    // Recieves an update from the subject:
    public function Update($symbol, $price)
    {
        if($this->symbol === $symbol)
        {
            $this->lastPrice = $price;
        }
    }

    // Used to get the most recent update information:
    public function GetTicker()
    {
        return "{$this->symbol} : {$this->lastPrice}";
    }
}

class StockTicker
{
    // The subscribed observers:
    private $observers = array();

    function __construct()
    {   }

    // Adds an observer to the list:
    public function AttachObserver($observer)
    {
        $this->observers[] = $observer;
    }

    // Notifies all observers:
    public function Notify($symbol, $price)
    {
        foreach($this->observers as $observer)
        {
            $observer->Update($symbol, $price);
        }
    }

}

File: Python/DesignPatterns/Observer.py
# Observer demo

# The Observer class that recieves updates from the ticker on a specific symbol:
class TickerObserver(object):

    def __init__(self, tickerSymbol):
        self.tickerSymbol = tickerSymbol
        self.lastPrice = 0.0

    def Update(self, symbol, price):
        if symbol == self.tickerSymbol:
            self.lastPrice = price

    def GetTicker(self):
        return self.tickerSymbol + " : " + str(self.lastPrice)
    


# The ticker subject that sends out updates to all observers:
class StockTicker(object):

    def __init__(self):
        self.observers = []

    def AttachObserver(self, tickerObserver):
        self.observers.append(tickerObserver)
    
    def RemoveObserver(self, tickerObserver):
        self.observers.remove(tickerObserver)

    def Notify(self, symbol, price):
        for observer in self.observers:
            observer.Update(symbol, price)